File: Java/AssignCookies.java
import java.util.Arrays;

class AssignCookies {
    public int findContentChildren(int[] g, int[] s) {
        Arrays.sort(g);
        Arrays.sort(s);
        int i = 0, j = 0;

        while (i < g.length && j < s.length) {
            if (g[i] <= s[j]) {
                i++;
            }
            j++;
        }

        return i;
    }

    public static void main(String[] args) {
        AssignCookies obj = new AssignCookies();
        int[] g = {1, 2, 3};
        int[] s = {1, 1};
        System.out.println(obj.findContentChildren(g, s));  // Output: 1
    }
}

File: LeetCode/Python/AssignCookies.py
from typing import List
class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        g.sort()
        s.sort()
        i = j = 0
        
        while i < len(g) and j < len(s):
            if g[i] <= s[j]:
                i += 1
            j += 1
        
        return i


obj=Solution()
g = [1,2,3]
s = [1,1]
print(obj.findContentChildren(g,s))